src/main/java/com/exasol/adapter/dialects/postgresql/PostgreSQLIdentifierMapping.java

```diff
@@ -33,4 +53,45 @@ public static PostgreSQLIdentifierMapping parse(final String mapping) {
                     .message("Unable to parse PostgreSQL identifier mapping from a null value.").toString());
         }
     }
+
+    /**
+     * Read identifier mapping from adapter properties.
+     *
+     * @param properties adapter properties to read identifier mapping from
+     * @return identifier mapping from properties or default value
+     */
+    public static CaseFolding from(final AdapterProperties properties) {
+        return properties.containsKey(PROPERTY) //
+                ? CaseFolding.valueOf(properties.get(PROPERTY))
+                : CaseFolding.DEFAULT;
+    }
+
+    /**
+     * @return validator for adapter properties controlling identifier mapping
+     */
+    public static PropertyValidator validator() {
+        return new Validator();
+    }
+
+    static class Validator implements PropertyValidator {
+        @Override
+        public void validate(final AdapterProperties properties) throws PropertyValidationException {
+            if (properties.containsKey(PROPERTY)) {
+                final String propertyValue = properties.get(PROPERTY);
+                if (!propertyValue.equals(PRESERVE_ORIGINAL_CASE) && !propertyValue.equals(CONVERT_TO_UPPER)) {
+                    throw new PropertyValidationException(ExaError.messageBuilder("E-VSPG-4") //
+                            .message("Value for {{property}} must be {{value_1}} or {{value_2}}.", PROPERTY,
+                                    PRESERVE_ORIGINAL_CASE, CONVERT_TO_UPPER) //
```

Comment:
In my opinion it doesn't make sense to use placeholders for the two values. Just concatenate the string.
But you can leave it like this if you prefer :)